packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_INPUT_TYPE,
+  OdsInputValueChangeEventDetail,
+  OsdsInputCustomEvent,
+} from '@ovhcloud/ods-components';
+import {
+  OsdsButton,
+  OsdsInput,
+  OsdsModal,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import { useTranslation } from 'react-i18next';
+
+type TEditNameModal = {
+  oldName: string;
+  onClose: () => void;
+  onConfirm: (newName: string) => void;
+};
+
+export default function EditNameModal({
+  oldName,
+  onClose,
+  onConfirm,
+}: Readonly<TEditNameModal>) {
+  const { t } = useTranslation('edit-name');
+  const [newName, setNewName] = useState(oldName);
+
+  const isValidName = /^[a-z0-9][-_.A-Za-z0-9]{1,61}[a-z0-9]$/.test(newName);
```

Comment:
Maybe this one could be a dedicated utility function with a dedicated test suite, what do you think?